Stratis.Bitcoin/BlockStore/BlockRepository.cs

```diff
@@ -293,40 +453,203 @@ public Task DeleteAsync(uint256 newlockHash, List<uint256> hashes)
+        {
+            var addrRow = this.session.Transaction.Select<byte[], byte[]>("Address", addr);
+            if (addrRow.Exists)
+            {
+                // Get the number of transactions
+                var cntBytes = addrRow.GetValuePart(0, sizeof(uint));
+                if (!BitConverter.IsLittleEndian) Array.Reverse(cntBytes);
+                uint cnt = BitConverter.ToUInt32(cntBytes, 0);
+
+                // Any transactions recorded?
+                if (cnt > 0)
+                {
+                    const int elementSize = 36;
+
+                    byte[] listBytes = addrRow.GetValuePart(sizeof(uint), (uint)(cnt * elementSize));
+                    for (int i = 0; i < cnt; i++)
+                    {
+                        var trxid = new byte[32];
+                        Array.Copy(listBytes, i * elementSize, trxid, 0, elementSize);
+                        yield return trxid;
+                    }
+                }
+            }
+        }
+
+        private void RecordTransactionOutputsToAddress(uint160 addr, HashSet<byte[]> list)
+        {
+            uint cnt = 0;
+            var addrRow = this.session.Transaction.Select<byte[], byte[]>("Address", addr.ToBytes());
+
+            if (addrRow.Exists)
+            {
+                // Get the number of transaction hashes
+                var bytes = addrRow.GetValuePart(0, sizeof(uint));
+                if (!BitConverter.IsLittleEndian) Array.Reverse(bytes);
+                cnt = BitConverter.ToUInt32(bytes, 0);
+            }
+
+            const int elementSize = 36; // Transaction Hash + output number = sizeof(trxid) + sizeof(uint)
+
+            // Force the value to a power of 2 that will fit all the additional transaction hashes
+            if ((cnt + list.Count) >= 2 && (addrRow.Value == null || (sizeof(uint) + (cnt + list.Count) * elementSize) > addrRow.Value.Length))
+            {
+                uint growCnt = 2;
+                for (; growCnt < (cnt + list.Count); growCnt *= 2) { }
+                uint growSize = sizeof(uint) + growCnt * elementSize; 
+                this.session.Transaction.InsertPart<byte[], byte[]>("Address", addr.ToBytes(), new byte[] { 0 /* Dummy */}, growSize - 1);
+            }
+
+            // Add the transaction hashes
+            foreach (var trxId in list)
+                this.session.Transaction.InsertPart<byte[], byte[]>("Address", addr.ToBytes(), trxId.ToBytes(), sizeof(uint) + (cnt++) * elementSize);
+
+            // Update the number of transaction hashes
+            var cntBytes = BitConverter.GetBytes(cnt);
+            if (!BitConverter.IsLittleEndian) Array.Reverse(cntBytes);
+            
+            this.session.Transaction.InsertPart<byte[], byte[]>("Address", addr.ToBytes(), cntBytes, 0);
+        }
+
+        private int RemoveTransactionOutputsToAddress(uint160 addr, HashSet<byte[]> outputs)
+        {
+            const int elementSize = 36;
```

Comment:
I have added a small helper class that defines the constant TransactionOutputID.Length = 36.